src/conf_mode/system-login.py

```diff
@@ -64,6 +64,11 @@ def get_config(config=None):
     login = conf.get_config_dict(base, key_mangling=('-', '_'),
                                  no_tag_node_value_mangle=True, get_first_key=True)
 
+    # Get global OTP parameters
+    default_values_otp = defaults(base + ['authentication'])
+    login['authentication'] = {}
```

Comment:
Why override this part of the config with an empty dict before merging default values?

Reply:
Without it the script will display an error that this key does not exist:
```
Traceback (most recent call last):
  File "/usr/libexec/vyos/conf_mode/system-login.py", line 339, in <module>
    c = get_config()
  File "/usr/libexec/vyos/conf_mode/system-login.py", line 70, in get_config
    login['authentication'] = dict_merge(default_values_otp, login['authentication'])
KeyError: 'authentication'
```

Reply:
Should add a check if it doesn't exist, If authentication node does exist in the config it would be cleared leaving only default values.
```
if 'authentication' not in login:
    login['authentication'] = {}
```

Reply:
Actually @goodNETnick and @sarthurdev all is already there and should follow the style already used. In this case:

```python
if 'authentication' in login:
    default_values = defaults(base + ['authentication'])
    login['authentication'] = dict_merge(default_values, login['authentication'])
```

Reply:
The actual common style used in all implementations is:

```python
# retrieve default values
default_values = defaults(base)

# XXX: T2665: When there is no FOO configuration given, we can safely
# remove the default values from the dict.
if 'user' not in login:
    if 'user' in default_values:
        del default_values['user']

# ... same goes for radius ...
if RADIUS-FOO not in login ....
```

Reply:
Thanks! Fixed

Reply:
Is it fixed now?

Reply:
Why not use the solution I provided? Is there something wrong with it?

Reply:
I think your solution is great.
But in this case the default values are required not only for the **login authentication** section, but also for the **login user name authentication** section.
In that case, this solution does not look bad either.
Or is it just as bad as the original one?